### PR TITLE
Fix some bugs and code style issues

### LINKED ISSUE #132: Not all commands are shown in `help`
help apply (no help)
help addfindrule (no help)
help deletefindrule (no help)
help opentab (no help)
help closetab (no help)
help rewind (no help)
help import
help export

### PR DESCRIPTION
**Pull request by le0tan**
Closes #132

### CODE REVIEW

File: src/main/java/seedu/algobase/ui/ListElementPointer.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    /**
+     * Returns the next element in the list and advances the cursor position.
+     * @throws NoSuchElementException if there is no more next element in the list.
+     */
+    public String next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+        return list.get(++index);
```
Comment (tiuweehan) [inline src/main/java/seedu/algobase/ui/ListElementPointer.java:72]: Doesn't seem like good practice, though since it's from AB4 I guess it should be fine

**Review by tiuweehan — APPROVED** [review]
Looks fine to me

**Review by jiayushe — APPROVED** [review]
LGTM

**Comment by le0tan** [comment]
@jiayushe please check if my merging caused any trouble

File: src/main/java/seedu/algobase/logic/commands/Command.java [hunk 2]
```diff
@@ -24,20 +27,28 @@
...
         DeleteTaskCommand.class,
         DoneTaskCommand.class,
         EditCommand.class,
         EditPlanCommand.class,
         EditTagCommand.class,
         EditTaskCommand.class,
+        ExportCommand.class,
+        ImportCommand.class,
+        MoveTaskCommand.class,
+        OpenTabCommand.class,
```
Comment (jiayushe) [inline src/main/java/seedu/algobase/logic/commands/Command.java]: `MoveTask` repeats with below. Try to arrange them in alphabetical order.

File: src/main/java/seedu/algobase/logic/commands/task/DeleteTaskCommand.java [hunk 3]
```diff
@@ -49,28 +51,22 @@ public DeleteTaskCommand(DeleteTaskDescriptor deleteTaskDescriptor) {
...
         }
-        Task task = taskList.get(taskIndex);
-        taskList.remove(taskIndex);
-        Set<Task> taskSet = new HashSet<>(taskList);
 
+        Plan planToUpdate = lastShownPlanList.get(deleteTaskDescriptor.planIndex.getZeroBased());
+        List<Task> taskList = new ArrayList<>(planToUpdate.getTasks());
+        Task task = taskList.get(deleteTaskDescriptor.taskIndex.getZeroBased());
+        taskList.remove(deleteTaskDescriptor.taskIndex.getZeroBased());
+        Set<Task> taskSet = new HashSet<>(taskList);
```
Comment (jiayushe) [inline src/main/java/seedu/algobase/logic/commands/task/DeleteTaskCommand.java]: Revert changes in `execute`.

File: src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java [hunk 4]
```diff
@@ -105,34 +106,35 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
...
-        assertCommandSuccess(command, actualModel, expectedCommandResult, expectedModel);
+        assertCommandSuccess(command, actualModel, actualCommandHistroy, expectedCommandResult, expectedModel);
     }
 
     /**
      * Executes the given {@code command}, confirms that <br>
      * - a {@code CommandException} is thrown <br>
      * - the CommandException message matches {@code expectedMessage} <br>
-     * - the algobase, filtered problem list and selected problem in {@code actualModel} remain unchanged
+     * - the address book, filtered person list and selected person in {@code actualModel} remain unchanged
```
Comment (jiayushe) [inline src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java]: Revert this line.

File: src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java [hunk 5]
```diff
@@ -105,34 +106,35 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
...
      */
-    public static void assertCommandFailure(Command command, Model actualModel, String expectedMessage) {
+    public static void assertCommandFailure(Command command, Model actualModel, CommandHistory actualCommandHistroy,
+                                            String expectedMessage) {
         // we are unable to defensively copy the model for comparison later, so we can
         // only do so by copying its components.
-        AlgoBase expectedAlgoBase = new AlgoBase(actualModel.getAlgoBase());
+        AlgoBase expectedAddressBook = new AlgoBase(actualModel.getAlgoBase());
```
Comment (jiayushe) [inline src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java]: ^

File: src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java [hunk 6]
```diff
@@ -105,34 +106,35 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
...
         List<Problem> expectedFilteredList = new ArrayList<>(actualModel.getFilteredProblemList());
 
-        assertThrows(CommandException.class, expectedMessage, () -> command.execute(actualModel));
-        assertEquals(expectedAlgoBase, actualModel.getAlgoBase());
+        assertThrows(CommandException.class, expectedMessage, () -> command.execute(actualModel, actualCommandHistroy));
+        assertEquals(expectedAddressBook, actualModel.getAlgoBase());
```
Comment (jiayushe) [inline src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java]: ^

File: src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java [hunk 7]
```diff
@@ -105,34 +106,35 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
...
         assertEquals(expectedFilteredList, actualModel.getFilteredProblemList());
     }
     /**
      * Updates {@code model}'s filtered list to show only the person at the given {@code targetIndex} in the
-     * {@code model}'s algobase.
+     * {@code model}'s address book.
```
Comment (jiayushe) [inline src/test/java/seedu/algobase/logic/commands/CommandTestUtil.java]: ^

**Review by jiayushe — CHANGES_REQUESTED** [review]
LGTM. Just revert some unnecessary changes.

File: src/main/java/seedu/algobase/logic/commands/task/DeleteTaskCommand.java (hunk 3 above)
Comment (le0tan) [reply src/main/java/seedu/algobase/logic/commands/task/DeleteTaskCommand.java]: What does this mean?

Comment (jiayushe) [reply src/main/java/seedu/algobase/logic/commands/task/DeleteTaskCommand.java]: Hmm revert back to my version. I checked for index out of bound for tasks.

**Review by jiayushe — APPROVED** [review]
LGTM

**Pull request merged**